src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -387,6 +387,13 @@ public override ConstantValue ConstantValue
         {
             get { return this.ConstantValueOpt; }
         }
+
+        public override NullabilityInfo TopLevelNullabilityInfo
```

Comment:
>TopLevelNullabilityInfo [](start = 40, length = 23)

Can we remove the override and seal the definition in BoundNode?

Reply:
This is working around an issue that Neal has fixed in master that I haven't checked to ensure is resolved in my branch yet: you couldn't have a TSWA with a null symbol that wasn't considered Unknown. That may be fixed, but I haven't checked it yet.

As to the `BoundNode` vs `BoundExpression`, I specifically did this because only things that derive from `BoundExpression` should be given a nullability. The info needs to be stored in the `BoundNode` because that's where the flags live, but it shouldn't be possible to set nullability on anything that isn't a `BoundExpression`.

---
In reply to: [264938565](https://github.com/dotnet/roslyn/pull/34017#discussion_r264938565) [](ancestors = 264938565)

Reply:
Please add a PROTOTYPE comment to make the `BoundExpression` implementation non-virtual.

---
In reply to: [265247636](https://github.com/dotnet/roslyn/pull/34017#discussion_r265247636) [](ancestors = 265247636,264938565)

Reply:
nit: comment could be more specific to help remind to remove this property

---
In reply to: [265667308](https://github.com/dotnet/roslyn/pull/34017#discussion_r265667308) [](ancestors = 265667308,265247636,264938565)

Reply:
There's a comment on BoundExpression to make the property non-virtual.

---
In reply to: [265805034](https://github.com/dotnet/roslyn/pull/34017#discussion_r265805034) [](ancestors = 265805034,265667308,265247636,264938565)